providers/implementations/rands/build.info

```diff
@@ -3,4 +3,4 @@ SUBDIRS=seeding
 $COMMON=drbg.c test_rng.c drbg_ctr.c drbg_hash.c drbg_hmac.c crngt.c rand_pool.c
 
 SOURCE[../../libfips.a]=$COMMON
-SOURCE[../../libnonfips.a]=$COMMON
+SOURCE[../../libnonfips.a]=$COMMON fake.c
```

Comment:
Ah, now I understand.  This seems misplaced.

I would rather suggest moving that file to `test/rand_fake.c` and use it directly as source for the programs that need it.

Reply:
but the file needs access to ```drbg_local.h``` which is in that same directory.  Still recommend that?  How should I rewrite the include, if so?

Reply:
Why does it need access to `drbg_local.h`?
RNGs can be done without this.  See `providers/implementations/rands/seed.c` in #13226 e.g.

Reply:
I didn't know details of #13226, and I don't see how you "swap out" the existing RNG unless it's a DRBG.  So using that PR's ```seed.c``` will require something other than the swap functions. I don't know how to do that "something else" other than implementing a private internal ```RAND_METHOD```

Reply:
You should end up with an EVP_RAND_CTX for the new source.  It doesn't have to be a DRBG, and provider side RNG can be wrapped by EVP_RAND.  It might require some provider infrastructure though -- maybe an alternative for the EVP_X_meth_ calls (which don't mandate an engine) is something we could to look into supporting.  `EVP_RAND_from_dispatch()`?

I might have missed some of the essential parameters -- the provider side should be good, I've not yet properly checked what EVP demands.

Reply:
That might be a cleaner way of doing this, but it would require more time (and more importantly, knowledge) than I have.